### PR TITLE
decomp hud functions

### PR DESCRIPTION
**Pull request by Rainchus**

### CODE REVIEW

File: src/d5a50_len_5fd0.c [hunk 1]
```diff
@@ -46,21 +46,33 @@ INCLUDE_ASM(s32, "d5a50_len_5fd0", render_hud_elements_world);
...
+}
 
-INCLUDE_ASM(s32, "d5a50_len_5fd0", func_80144258);
+void func_80144258(s32 arg0) {
+    func_80143C48(arg0, 1, 1);
+}
 
-INCLUDE_ASM(s32, "d5a50_len_5fd0", func_80144278);
+void func_80144278(s32 arg0) {
+    func_80143C48 (arg0, 0, 1);
```
Comment (ethteck) [inline src/d5a50_len_5fd0.c:62]: space here between func name and opening paren

File: src/d5a50_len_5fd0.c [hunk 2]
```diff
@@ -114,7 +126,13 @@ INCLUDE_ASM(s32, "d5a50_len_5fd0", set_hud_element_size);
 
 INCLUDE_ASM(s32, "d5a50_len_5fd0", func_80144E4C);
 
-INCLUDE_ASM(s32, "d5a50_len_5fd0", func_80144E74);
+void func_80144E74(s32 iconIndex, s32 arg1) {
+    HudElement *hudElement;
```
Comment (ethteck) [inline src/d5a50_len_5fd0.c:130]: Pointer should be next to type name (HudElement*) and also you can initialize hudElement on the same line: `HudElement* hudElement = gHudElementList[iconIndex & ~0x800];`

**Review by ethteck — CHANGES_REQUESTED** [review]
grood

**Review by ethteck — APPROVED** [review]

File: src/d5a50_len_5fd0.c [hunk 3]
```diff
@@ -46,21 +46,33 @@ INCLUDE_ASM(s32, "d5a50_len_5fd0", render_hud_elements_world);
...
 
-INCLUDE_ASM(s32, "d5a50_len_5fd0", func_80144278);
+void func_80144278(s32 arg0) {
+    func_80143C48(arg0, 0, 1);
+}
 
 INCLUDE_ASM(s32, "d5a50_len_5fd0", draw_hud_element);
 
-INCLUDE_ASM(s32, "d5a50_len_5fd0", draw_hud_element_clipped);
+void draw_hud_element_clipped (s32 arg0) {
```
Comment (nanaian) [inline src/d5a50_len_5fd0.c:67]:
```suggestion
void draw_hud_element_clipped(s32 arg0) {
```

**Review by nanaian — CHANGES_REQUESTED** [review]

**Pull request merged**